definitions/marts/looker_studio_marts/ecf/ls_ecf_provider_activity_schedule_changes.sqlx

```diff
@@ -0,0 +1,117 @@
+      event_date_delta: "The number of days between the event_date and the current date.",
+      cpd_lead_provider_name: "The name of the lead provider who performed the action. This is determined by the request_user_id which is captured in the service and mapped against the cpd_lead_provider_id in cpd_lead_providers_latest. This connection has been confirmed as correct by LPDOB.",
+	  request_uuid: "The unique identifier for the API request that triggered the change to the participant's schedule_id. This UUID is only usable by LPDOB or developers if the request needs to be reviewed. In this mart this value is unique and a distinct count of this field is used to determine how many actions of this type have been conducted by a lead provider on the event date.",
+	  participant_profile_id: "The participant profile ID of the participant whose schedule has been changed.",
+	  user_id: "The user ID taken from the teacher profile associated with the participant profile.",
+	  participant_type: "The type of the participant whose schedule has been changed. The value can be either ECT or Mentor.",
+	  request_path: "The full API path used by the lead provider to change the participant's schedule. The path contains the user_id but always ends with 'change-schedule'.",
+	  key_updated: "The name of the field in the table participant_profiles_latest_cpd that was changed by the API request. This is always schedule_id as that is the only field a lead provider can amend via the 'change-schedule' API path.",
+	  new_schedule: "The new value for the schedule_id field on the participant profile translated into text via schedules_latest_cpd.",
+	  new_cohort: "The cohort associated with the new schedule. Even if the cohort has not changed this field will be populated with the cohort that is associated with the new schedule.",
+	  previous_schedule: "The previous value for the schedule_id field on the participant profile translated into text via schedules_latest_cpd.",
+	  previous_cohort: "The cohort associated with the previous schedule. Even if the cohort has not changed this field will be populated with the cohort that is associated with the previous schedule."
+    }
+}
+
+-- Gather ECF Induction Records to be able to join User ID, Participant Type & Cohort onto participants within the script without needing to join on Participant Profiles and Teacher Profiles separately.
+WITH ecf_inductions AS(
+  SELECT
+     induction_record_id
+    ,participant_profile_id
+    ,participant_type
+    ,user_id
+    ,cohort
+    ,DATE(start_date) AS start_date
+    ,IFNULL(DATE(end_date), DATE(2050,12,31)) AS end_date
+  FROM
+    ${ref('ecf_inductions')}
+)
+
+-- Gather Schedule Changes made via the API by Lead Providers. These API changes impact the participant profile and as such we take the information from the field changes made to this table. When a schedule changes so can the cohort and in this CTE we capture both values in their new and previous states.

```

Comment:
Is it the schedule change that prompts the cohort change, or do they just happen simultaneously? If it's the former, then it might be worth saying: "Schedule changes can trigger cohort changes ..." if it's the latter, "Cohort changes can occur simultaneously with schedule changes so .."

Reply:
Each schedule has an associated cohort. If the new schedule's cohort is different to the old schedules cohort then the cohort has changed.

Reply:
Okay, can you change to the recommendation above so that it's clear? "Schedule changes can trigger cohort changes ..."

Reply:
Have reworded "When a schedule changes so can the cohort" to "Schedule changes can trigger cohort changes"